### ISSUE TITLE
Cluster randomly crashed?

### ISSUE THREAD

**Issue by ingshtrom** [coreos/tectonic-forum#152]
### Tectonic Version 
v1.6.4

### Environment

What hardware/cloud provider/hypervisor is being used with Tectonic?
AWS

### Expected Behavior
it doesn't crash? hah.

### Actual Behavior
It crashed.. i think.

### Reproduction Steps
no idea, honestly.

### Other Information
Alright, so I have no idea, but here is what happened. Yesterday (6/26) I created the k8s cluster with Tectonic. Since then I've been working on getting Istio working correctly. I got it working today around 2pm and haven't touched it yet. At that time, the BookInfo sample for Istio was working. I got home and then around 8:30pm I logged on to have been logged out of Tectonic Console and the BookInfo sample was no longer working. 

After this I tried refreshing the login page and nothing was happening. I was getting timeouts every time. I starting trying to figure out what happened, so I ssh'ed into a master node and ran an etcdctl command to check the health of the etcd cluster. I ran `etcdctl --endpoints=<etcd_node_private_ip> cluster-health` and got this:
![image](https://user-images.githubusercontent.com/571756/27616542-763b6db4-5b7e-11e7-97a3-affd7ac04255.png)

Ping also failed to that IP. I triple checked it as well--it's correct in the AWS EC2 console. So etcd must be down right?... well I'm not sure. I didn't know what to do, and then I randomly refreshed the tectonic console... it loads... what the heck!?

I check the `etcd cluster-health` command again and it still fails. So does pinging... so I guess just chalk that up to voodoo magic? idk.

Anywho, it's working and I'm trying to figure out why. When I look at the k8s events in the Tectonic Console, it stops at 8:29pm... which is right around the time I first tried to access it. All of the events after that show that all of the previous containers are being created fresh... what is that? why? who? what? I don't even know what I should be asking. 

![image](https://user-images.githubusercontent.com/571756/27616708-9a1a1932-5b7f-11e7-8944-f9d225724cca.png)

To verify that the cluster was created yesterday, in AWS I double checked the launch date and it was yesterday.
![image](https://user-images.githubusercontent.com/571756/27616736-e5752e30-5b7f-11e7-964f-f0b5fb242fb0.png)

So in conclusion,  the cluster crashed at some point and then fixed itself once I started poking it?  I would love any help. I'm new to Kubernetes so I'm not event sure where to start.

Thank you.

**Comment by ingshtrom** [comment]
hmmm. I guess these event logs just don't go very far back in time, because now it says...

![image](https://user-images.githubusercontent.com/571756/27617281-5af02fae-5b83-11e7-9e89-2581d157998d.png)

**Comment by robszumski** [comment]
Running down the initial list of thoughts:
1. etcd machines are large enough instances, shouldn't be an issue with disk IO or anything

> When I look at the k8s events in the Tectonic Console, it stops at 8:29pm

2. If there are no events they won't be shown. This isn't always a bad thing, if nothing is going on in the cluster.

> hmmm. I guess these event logs just don't go very far back in time, because now it says...

This is unfortunately true, Kubernetes doesn't store these very long. They are good for short term troubleshooting however. You can [opt to send them](https://coreos.com/tectonic/docs/latest/admin/logging.html) to a longer term storage solution if you'd like.

3. Since you didn't lose your etcd state, I don't think any quorum issues happened. You could also jump onto one of the etcd boxes and check the journal there to confirm. This will probably have the best info to continue debugging.

> All of the events after that show that all of the previous containers are being created fresh... what is that?

4. This makes me think that a node was marked as `NotReady` and the containers were rescheduled to other nodes in the cluster. We are triaging [an issue](https://github.com/coreos/tectonic-installer/pull/1224) possibly related. This wouldn't have affected your etcd machines or the communication to them, however.

In summary, some more investigation needs to happen, but I don't think the cluster "crashed", but some workloads were shuffled around between machines. Grab some details from the etcd log and we can see where to go next.

**Comment by ingshtrom** [comment]
Alright, I'm actually going to close this because I had to kill the cluster >< Thank you for the response and I will update with that info if I see it again.
